website/scripts/optimize-images.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * Image optimization script
+ *
+ * Automatically converts images to modern formats (WebP, AVIF) with responsive sizes.
+ * Runs during the build process to optimize images without requiring manual intervention.
+ *
+ * Features:
+ * - Converts PNG/JPEG to WebP (85% quality) and AVIF (80% quality)
+ * - Generates responsive sizes matching Tailwind CSS breakpoints
+ * - Outputs to static/optimized-images for both production and development
+ * - Only processes new/changed images (incremental optimization)
+ * - Intelligent filtering of already optimized images
+ */
+
+const sharp = require('sharp');
+const glob = require('glob');
+const path = require('path');
+const fs = require('fs').promises;
+
+async function optimizeImages() {
```

Comment:
should be covered by unit tests and written in typescript

Reply:
You're right, and I am sorry about it.